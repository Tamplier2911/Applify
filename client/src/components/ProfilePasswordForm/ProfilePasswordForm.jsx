// import "./ProfilePasswordForm.scss";
import React, { useState } from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";

// components
import FormInput from "../FormInput/FormInput";
import Button from "../Button/Button";

// JS Rendering CSS
import {
  ProfilePasswordFormContainer,
  ProfilePasswordFormTitle,
  ProfilePasswordFormElement
} from "./ProfilePasswordFormStyles";

// component constants
import profilePasswordFormData from "../../utils/ComponentProfilePasswordFormConstants/componentProfilePasswordFormConstants";

const ProfilePasswordForm = ({ lang }) => {
  const {
    profilePasswordFormTitle,
    profilePasswordFormCurPass,
    profilePasswordFormNewPass,
    profilePasswordFormNewPassConf,
    profilePasswordFormFileButton
  } = profilePasswordFormData[lang];

  const [userPassword, setUserPassword] = useState({
    currentPassword: "",
    newPassword: "",
    newPasswordConfirm: ""
  });

  const { currentPassword, newPassword, newPasswordConfirm } = userPassword;

  const onInputChange = e => {
    const { name, value } = e.target;
    setUserPassword({ ...userPassword, [name]: value });
  };

  const onFormSubmit = e => {
    e.preventDefault();
    // validate pass
    console.log(userPassword);
  };

  // console.log(userPassword, "from ProfilePasswordForm");

  return (
    <ProfilePasswordFormContainer>
      <ProfilePasswordFormTitle lang={lang}>
        {profilePasswordFormTitle}
      </ProfilePasswordFormTitle>
      <ProfilePasswordFormElement
        autoComplete="off"
        onSubmit={e => onFormSubmit(e)}
      >
        <FormInput
          onInputChange={e => onInputChange(e)}
          value={currentPassword}
          label={profilePasswordFormCurPass}
          name="currentPassword"
          type="password"
          required
        />
        <FormInput
          onInputChange={e => onInputChange(e)}
          value={newPassword}
          label={profilePasswordFormNewPass}
          name="newPassword"
          type="password"
          required
        />
        <FormInput
          onInputChange={e => onInputChange(e)}
          value={newPasswordConfirm}
          label={profilePasswordFormNewPassConf}
          name="newPasswordConfirm"
          type="password"
          required
        />
        <Button type="submit" value={profilePasswordFormFileButton} />
      </ProfilePasswordFormElement>
    </ProfilePasswordFormContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  lang: selectCurrentLanguage
});

export default connect(mapStateToProps)(ProfilePasswordForm);
