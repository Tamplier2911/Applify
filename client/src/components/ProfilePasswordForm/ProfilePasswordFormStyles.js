import styled from "styled-components";
import ScrollAnimation from "react-animate-on-scroll";

const getFont = (props) => {
  const { lang } = props;
  if (lang === "eng") {
    return `font-family: var(--font-title);`;
  } else if (lang === "rus" || lang === "ukr") {
    return `font-family: var(--font-cyrillic);`;
  }
  return `font-family: var(--font-title);`;
};

export const ProfilePasswordFormContainer = styled(ScrollAnimation)`
  display: grid;
  grid-row-gap: 4rem;
  grid-auto-rows: min-content;

  padding: 2rem;
  box-shadow: 0 0.1rem 0.5rem var(--cl-font);

  @media only screen and (max-width: 768px) {
    margin-bottom: 5rem;
  }
`;

export const ProfilePasswordFormTitle = styled.h3`
  ${getFont}
  font-weight: 400;
  font-size: 1.7rem;

  margin-top: 2rem;
  @media only screen and (max-width: 425px) {
    //   margin-top: 5rem;
  }
`;

export const ProfilePasswordFormElement = styled.form`
  display: grid;
  grid-row-gap: 2rem;
`;
