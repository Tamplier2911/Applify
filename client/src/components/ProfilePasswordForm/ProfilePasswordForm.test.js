import React from "react";

// component
import ProfilePasswordForm from "./ProfilePasswordForm";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
} from "../../tests/testUtils";

describe("<ProfilePasswordForm />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };
  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<ProfilePasswordForm store={store} />);
  });

  it("renders without an error", () => {
    const profilePasswordForm = findByTestAttr(
      wrapper,
      "profile-password-form"
    );
    expect(profilePasswordForm.length).toBe(1);
  });

  it("renders - children profile info form element", () => {
    const profilePasswordFormElement = findByTestAttr(
      wrapper,
      "profile-password-form-element"
    );
    expect(profilePasswordFormElement.length).toBe(1);
  });

  it("matches snapshot", () => {
    const profilePasswordForm = findByTestAttr(
      wrapper,
      "profile-password-form"
    );
    expect(profilePasswordForm).toMatchSnapshot();
  });
});
