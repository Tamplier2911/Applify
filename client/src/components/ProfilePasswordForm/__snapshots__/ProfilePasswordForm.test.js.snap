// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfilePasswordForm /> matches snapshot 1`] = `
<Styled(ScrollAnimation)
  animateIn="bounceInLeft"
  animateOnce={true}
  data-test="profile-password-form"
>
  <styled.h3
    lang="eng"
  >
    Change Password:
  </styled.h3>
  <styled.form
    autoComplete="off"
    data-test="profile-password-form-element"
    onSubmit={[Function]}
  >
    <FormInput
      label="Current Password"
      name="currentPassword"
      onInputChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <FormInput
      label="New Password"
      name="newPassword"
      onInputChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <FormInput
      label="Confirm New Password"
      name="newPasswordConfirm"
      onInputChange={[Function]}
      required={true}
      type="password"
      value=""
    />
    <Button
      type="submit"
      value="Change Password"
    />
  </styled.form>
</Styled(ScrollAnimation)>
`;
